packages/opal-server/opal_server/scopes/api.py

```diff
@@ -53,10 +57,14 @@ def _check_worker_token(authorization: str = Header()):
         if scheme.lower() != "bearer" or token != opal_server_config.WORKER_TOKEN:
             raise Unauthorized()
 
-    @router.put(
-        "", status_code=status.HTTP_201_CREATED, dependencies=[Depends(authenticator)]
-    )
-    async def put_scope(*, scope_in: Scope):
+    @router.put("", status_code=status.HTTP_201_CREATED)
+    async def put_scope(*, scope_in: Scope, claims: JWTClaims = Depends(authenticator)):
+        try:
+            require_peer_type(authenticator, claims, PeerType.datasource)
```

Comment:
This code block repeats itself couple of times.
Can u put it into the dependency somehow? or think of another way to reuse it?